File: app/src/main/java/com/boardgamegeek/sorter/YearPublishedAscendingSorter.java
package com.boardgamegeek.sorter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import com.boardgamegeek.R;

public class YearPublishedAscendingSorter extends YearPublishedSorter {
	public YearPublishedAscendingSorter(@NonNull Context context) {
		super(context);
	}

	@StringRes
	@Override
	public int getTypeResource() {
		return R.string.collection_sort_type_year_published_asc;
	}

	@StringRes
	@Override
	public int getSubDescriptionId() {
		return R.string.oldest;
	}
}

File: app/src/main/java/com/boardgamegeek/sorter/YearPublishedAscendingSorter.kt
package com.boardgamegeek.sorter

import android.content.Context
import androidx.annotation.StringRes

import com.boardgamegeek.R
import com.boardgamegeek.entities.CollectionItemEntity

class YearPublishedAscendingSorter(context: Context) : YearPublishedSorter(context) {
    @StringRes
    public override val typeResId = R.string.collection_sort_type_year_published_asc

    @StringRes
    public override val subDescriptionResId = R.string.oldest

    override fun sort(items: Iterable<CollectionItemEntity>) = items.sortedBy { it.gameYearPublished }
}
